velox/exec/PrefixSort.cpp

```diff
@@ -118,27 +130,28 @@ compareByWord(uint64_t* left, uint64_t* right, int32_t bytes) {
 
 PrefixSortLayout PrefixSortLayout::makeSortLayout(
     const std::vector<TypePtr>& types,
+    const std::vector<int32_t>& maxVarLengths,
     const std::vector<CompareFlags>& compareFlags,
     uint32_t maxNormalizedKeySize) {
   uint32_t normalizedKeySize = 0;
   uint32_t numNormalizedKeys = 0;
   const uint32_t numKeys = types.size();
   std::vector<uint32_t> prefixOffsets;
+  std::vector<uint32_t> encodeSizes;
   std::vector<PrefixSortEncoder> encoders;
 
   // Calculate encoders and prefix-offsets, and stop the loop if a key that
   // cannot be normalized is encountered.
   for (auto i = 0; i < numKeys; ++i) {
-    if (normalizedKeySize > maxNormalizedKeySize) {
```

Comment:
Why change here?

Reply:
Before we check this here, there is still chance the normalizedKeySize  + encodedSize > maxNormalizedKeySize, I moved this check to line 149 to make sure `normalizedKeySize`  would not exceed the `maxNormalizedKeySize`.

Reply:
Yeah, thanks for catching it. Could you add a test for it?

Reply:
Test case `checkMaxNormalizedKeySizeForMultipleKeys` is added for this.